packages/docregistry/src/context.ts

```diff
@@ -681,7 +687,7 @@ export class Context<
   ): Promise<Contents.IModel> {
     const path = this._path;
     // Make sure the file has not changed on disk.
-    const promise = this._manager.contents.get(path, { content: false });
+    const promise = this._manager.contents.get(path, { content: true });
```

Comment:
I definitely would like to avoid pulling the content as comparing large file will take time and increase the wire load.

Is there a way to set better `this.contentsModel?.last_modified`?

Reply:
- this only happens when the modification time changes then it does a comparison of the files, it won't happen often
- we could add a comparison to files under a specific size and modification time difference

Reply:
Would it be possible to compute hashsum (e.g. md5) of the content on the server and on the frontend and compare the hashsums instead of comparing the contents directly?

Reply:
@fcollonval The problem with last_modified with fuse filesystems is not all of them handle the change/modified time correctly. If someone has their home directory mounted on a fuse filesystem this issue will constantly pop up, even though the file didn't change

Reply:
when the message comes up "File Changed" it is not technically correct, it should be "Timestamp Changed" because that is what the check is looking at, this would run a file change check

Reply:
@krassowski I can't seem to find where the API call is going to where the file is opened. I'd love to add in a md5sum check if you can point me in the right direction. Thanks!

Reply:
That would be jupyter-server file contents API see https://github.com/jupyter-server/jupyter_server/blob/main/jupyter_server/files/ and https://jupyter-server.readthedocs.io/en/latest/developers/contents.html. It would probably be good to start by opening an issue on jupyter-server repository to discuss

FYI we ship a `murmur2` function in the debugger package, so maybe we could reuse that for client-side check (if needed at all).

Reply:
Or we can compare the contents of the file on the server side, and if there are no changes, we don't operate the file, so that Timestamp doesn't change either